core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -931,6 +931,11 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
       }
       writer.endFunCall(date_diff);
       break;
+    case "HASHBUCKET":
+      if (call.getOperandList().size() > 0 && call.operand(0).getKind() != SqlKind.LITERAL) {
```

Comment:
can we check isEmpty instead of size

Reply:
what if its not literal 
call.operand(0).getKind()

Reply:
hashbucket can only take 3 types of argument 
1. hashbucket()
2. hashbucket(null)
3. hashbucket(hashrow(column))

Reply:
> can we check isEmpty instead of size

done the change